embedded-compositor/qml/main.qml

```diff
@@ -82,10 +83,10 @@ WaylandCompositor {
 
             Rectangle {
                 id: centerArea
-                anchors.top: topArea.bottom
-                anchors.left: leftArea.right
-                anchors.right: rightArea.left
-                anchors.bottom: bottomArea.top
+                anchors.top: !rootTransformItem.fullScreen ? topArea.bottom : rootTransformItem.top
+                anchors.left: !rootTransformItem.fullScreen ? leftArea.right : rootTransformItem.left
+                anchors.right: !rootTransformItem.fullScreen ? rightArea.left : rootTransformItem.right
+                anchors.bottom: !rootTransformItem.fullScreen ? bottomArea.top : rootTransformItem.bottom
```

Comment:
is the visible flag not sufficient ? Is it necessary to change the anchors ?

Reply:
The visible flag does only hide the Item, futhermore the item will no longer receive mouse events. 
https://doc.qt.io/qt-5/qml-qtquick-item.html#visible-prop 
https://doc.qt.io/qt-5/qml-qtquick-item.html#enabled-prop

The layout remains still the same, so it is necessary to set the anchors of the centerArea according to the `fullScreen`  property.